src/components/tagApplyMenu/AposTagApply.vue

```diff
@@ -169,78 +189,110 @@ export default {
           this.$refs.textInput.$el.querySelector('input').select();
         });
       } else {
+        // TODO: No current sign of `create-tag` usage. Delete if unused.
         this.$emit('create-tag', this.searchInputValue);
+
         const tag = {
           label: this.searchInputValue,
           slug: this.searchInputValue,
-          checked: this.applyTo
+          match: this.applyTo
         };
-        this.checkboxes[tag.slug] = createCheckbox(tag, this.applyTo);
+        this.checkboxes[tag.slug] = this.createCheckbox(tag, this.applyTo);
         this.myTags.unshift(tag);
         this.creating = false;
         this.emitState();
       }
     },
-    update(slug) {
+    // `checked` will track all that are checked, but `updateTag` also
+    // updates indeterminate state and `match` arrays for the updated tag.
+    updateTag($event) {
+      const slug = $event.target.value;
+      // Find the matching tag.
       const tag = this.myTags.find(tag => tag.slug === slug);
+      // Find the matching checkbox.
       const box = this.checkboxes[slug];
-      if (!tag.checked) {
-        tag.checked = this.applyTo;
-        box.value.data.push(slug);
+      if (!box) {
+        // This should never happen. You updated a checkbox to trigger this.
+        return;
+      }
+
+      if (!tag.match) {
+        // If the tag has no `match` values, assign all the `applyTo` IDs to it.
+        tag.match = this.applyTo;
       } else {
-        if (tag.checked.length === this.applyTo.length) {
+        if (tag.match.length === this.applyTo.length) {
+          // If the tag has existings matches
           // previously full check, unapply to all
-
-          delete tag.checked;
-          box.value.data = [];
-
+          delete tag.match;
         } else {
-          // mixed check, check all
-          tag.checked = this.applyTo;
-          this.checkboxes[slug].value.data = [slug];
-          delete box.status.readOnly;
+          // Previously indeterminate. Set it's matches to all of `applyTo`.
+          tag.match = this.applyTo;
+
+          delete box.status.indeterminate;
           delete box.choice.indeterminate;
         }
       }
-      // refresh checkboxes :}
+      // Force refresh the checkboxes.
       this.keyPrefix = this.generateId('key');
-      // done, emit
+
+      // TODO: This should probably have an "Apply" or "Save" button to confirm
+      // before running emitting the updates.
       this.emitState();
     },
     emitState() {
       this.$emit('update', this.myTags);
     },
+    // Take the string field value and get a lower case version.
     updateSearchInput(value) {
       this.searchInputValue = value.data.toLowerCase();
       if (!this.searchInputValue) {
         this.creating = false;
       }
-    }
-  }
-};
+    },
+    createCheckbox(tag, applyTo) {
```

Comment:
Making these methods on the component allows us to reference properties on `this`. It is making the function less pure, I admit.